lib/Sema/TypeCheckStmt.cpp

```diff
@@ -655,89 +632,11 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
-  SmallVector<LabeledStmt*, 2> ActiveLabeledStmts;
-
-  /// The destination block for a 'fallthrough' statement. Null if the switch
-  /// scope depth is zero or if we are checking the final 'case' of the current
-  /// switch.
-  /// FIXME: Only required because EnableASTScopeLookup can be false
-  CaseStmt /*nullable*/ *FallthroughSource = nullptr;
-  CaseStmt /*nullable*/ *FallthroughDest = nullptr;
-
   /// Skip type checking any elements inside 'BraceStmt', also this is
   /// propagated to ConstraintSystem.
   bool LeaveBraceStmtBodyUnchecked = false;
 
   ASTContext &getASTContext() const { return Ctx; };
-  
-  struct AddLabeledStmt {
-    StmtChecker &SC;
-    AddLabeledStmt(StmtChecker &SC, LabeledStmt *LS) : SC(SC) {
-      // Verify that we don't have label shadowing.
-      auto sourceFile = SC.DC->getParentSourceFile();
-      checkLabeledStmtShadowing(SC.getASTContext(), sourceFile, LS);
-
-      // In any case, remember that we're in this labeled statement so that
-      // break and continue are aware of it.
-      SC.ActiveLabeledStmts.push_back(LS);
-
-      // Verify that the ASTScope-based query for active labeled statements
-      // is equivalent to what we have here.
-      if (LS->getStartLoc().isValid() && sourceFile &&
-          SC.getASTContext().LangOpts.EnableASTScopeLookup &&
-          !SC.getASTContext().Diags.hadAnyError() &&
-          !SC.LeaveBraceStmtBodyUnchecked) {
-        // The labeled statements from ASTScope lookup have the
-        // innermost labeled statement first, so reverse it to
-        // match the data structure maintained here.
-        auto activeFromASTScope = ASTScope::lookupLabeledStmts(
-            sourceFile, LS->getStartLoc());
-        assert(activeFromASTScope.front() == LS);
-        std::reverse(activeFromASTScope.begin(), activeFromASTScope.end());
-        if (activeFromASTScope != SC.ActiveLabeledStmts) {
-          llvm::errs() << "Old: ";
-          llvm::interleave(SC.ActiveLabeledStmts, [&](LabeledStmt *LS) {
-            llvm::errs() << LS;
-          }, [&] {
-            llvm::errs() << ' ';
-          });
-          llvm::errs() << "\nNew: ";
-          llvm::interleave(activeFromASTScope, [&](LabeledStmt *LS) {
-            llvm::errs() << LS;
-          }, [&] {
-            llvm::errs() << ' ';
-          });
-          llvm::errs() << "\n";
-        }
-        assert(activeFromASTScope == SC.ActiveLabeledStmts);
-      }
-    }
-    ~AddLabeledStmt() {
-      SC.ActiveLabeledStmts.pop_back();
-    }
-  };
-  
-  struct AddSwitchNest {
-    StmtChecker &SC;
-    CaseStmt *OuterFallthroughSource;
-    CaseStmt *OuterFallthroughDest;
-    AddSwitchNest(StmtChecker &SC)
-      : SC(SC),
-        OuterFallthroughSource(SC.FallthroughSource),
-        OuterFallthroughDest(SC.FallthroughDest) {
-    }
-    
-    ~AddSwitchNest() {
-      SC.FallthroughSource = OuterFallthroughSource;
-      SC.FallthroughDest = OuterFallthroughDest;
-    }
-  };
```

Comment:
This is the old implementation of fall-through case statements. It's no longer needed since we can always use the new implementation. This change is also in a separate commit.